### ISSUE TITLE
No longer available on AMO

### ISSUE THREAD

**Issue by jawz101** [diegocr/CleanLinks#170]

**Comment by diegocr** [comment]
Yeah, they shut it down arguing "it doesn't work" :-/

I think i'll just upload the broken/experimental version i made with e10s support (that i kinda dislike by the reasons given in other ticket)

Actually a WebExtension version might be more suitable, but i don't have much time these days to develop it from scratch _again_... 🙄

**Comment by jawz101** [comment]
Well, I just appreciate your work. If anything, it feels like this add-on made for a good reference design for several concepts if anyone wanted to investigate controlling referrers.

Kinda wish you'd work on Privacy Badger :D  I get the feeling your ideas and interests in privacy and heuristics are good.

**Comment by leedoyle** [comment]
If they are going to do that with every extension that doesn't support e10s yet it's a tragedy...
> Yeah, they shut it down arguing "it doesn't work" :-/

Does an addon being not e10s automatically mean it's going to be deleted from AMO? If not, is there any procedure to challenge such allegations?

> I think i'll just upload the broken/experimental version i made with e10s support (that i kinda dislike by the reasons given in other ticket)

Looking forward to it, I've been enjoying your addon.

> but i don't have much time these days to develop it from scratch again

Good thing is that you'd only have to rewrite it once, because WebExtensions is there to stay.

**Comment by diegocr** [comment]
Thanks for the nice words @jawz101 :-)

@leedoyle 
> Good thing is that you'd only have to rewrite it once, ...

Well, that was also my thought went i spent a considerable amount of time rewriting the former restartless extension to support e10s 🙄

> Does an addon being not e10s automatically mean it's going to be deleted from AMO?

Potentially i'd say, but it's something we have to ask Mozilla...

> Looking forward to it, I've been enjoying your addon.

Thank you :)

**Comment by Bluscream** [comment]
Can you release XPI's on Github and make a tutorial how to install them or use the source?

**Comment by anewuser** [comment]
Former users of this extension can try [Redirector](https://github.com/einaregilsson/Redirector#download-links) instead. It doesn't come with pre-defined rules but the general purpose is the same.

@diegocr Instead of wasting your time rewriting the extension again, I suggest you and other people just to share their regexes on the Redirector page (maybe we could start a wiki for it?).

**Comment by Bluscream** [comment]
Can't you do both? I prefer CleanLinks because it is open source

**Comment by jawz101** [comment]
@anewuser to expound on that idea, it'd be nice to have a list hosted on github which was formatted to pull into the add-on.  That way we could all curate a list but not have to manually add them.

@Bluscream I'm pretty sure everything we've talked about _is_ open source

**Comment by anewuser** [comment]
@einaregilsson Has there been any progress on https://github.com/einaregilsson/Redirector/issues/30? I invited these people to take a look at Redirector, but they'd still be in the same boat in case the migration to WebExtension kills your work on the Firefox extension

@jawz101 I get what you mean, but @einaregilsson  and  @diegocr don't have time to work on something big like that. You'd need to able to do that by yourself or hire/convince other programmers. :)

@Gitoffthelawn Any remarks or words of wisdom on this opportunity to combine efforts with one another?

**Comment by einaregilsson** [comment]
@anewuser That issue was finished long ago. Redirector is now a WebExtensions addon. Although, because of a Mozilla bug there are now requests sent to the original url and the new url, and that bug is only scheduled to be fixed in FF55 unfortunately :(

**Comment by atomGit** [comment]
you have the best link cleaner i have found - thanks for your work and hopefully it will make it back to AMO in the future

have a good one

**Comment by Atavic** [comment]
@diegocr I keep my legacy FF with the addons that perfectly work, as **Clean Links**!
@leedoyle I have many addons as a back up from the Privacy/Security section in [AMO](https://addons.mozilla.org/en-US/firefox/extensions/privacy-security/) and will upload them if they disappear.
@Bluscream There's an [addon](https://github.com/diegocr/GitHubExtIns) for that, but you can do it yourself:

Download this repository, click on top right and choose _Download ZIP_.
An XPI file is just a rebranded ZIP file, so you must either rename the file type or create a new zipped file with thw same content, just save it with the .xpi extension instead of the default .zip one.

**Comment by leethax666** [comment]
came here super worried, glad it'll be released again soon!

**Comment by ng4ever** [comment]
Good thing I had a backup plan.

**Comment by alekksander** [comment]
what is it? ;) can be useful… got the feeling this addon won't stand up for mozilla's super new age demands…

**Comment by leethax666** [comment]
Redirector looks cool if someone wants to maintain a regex list or port whatever clean links uses

On May 2, 2017 3:13:39 PM PDT, alekksander <notifications@github.com> wrote:
>what is it? ;) can be useful… got the feeling this addon won't stand up
>for mozilla's super new age demands…
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/diegocr/CleanLinks/issues/170#issuecomment-298776805

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by alekksander** [comment]
is it still maintained? (doesn't seem to be)

**Comment by JanieBug74** [comment]
It's been dragging and droppind, I don;t know what to do

**Comment by pde** [comment]
@jawz101 Privacy Badger is interested in supporting link deobfuscation; we've [had an open ticket for this for a long time](https://github.com/EFForg/privacybadger/issues/85), but now have someone actually working on it.

Weirdly, to the extent that this functionality can be implemented by using regular expression rewrites of outbound request URLs, EFF's other major browser extension [HTTPS Everywhere](https://eff.org/httpse) may actually be a better fit, since it's basically a giant regular expression rewriting engine.

**Comment by Gitoffthelawn** [comment]
@pde For a customizable regular expression rewriting engine, you may want to take a look at Redirector: https://github.com/einaregilsson/Redirector

**Comment by alekksander** [comment]
how exactly does it work? does it do anything by default?

**Comment by Gitoffthelawn** [comment]
@alekksander It does nothing by default.  You create rules (wildcard or regex) and then it automatically acts on those rules.

**Comment by alekksander** [comment]
so standard rules from clean links works? ((?:ref|aff)\w*|utm_\w+|(?:merchant|programme|media)ID) how to set it up?
sorry for getting off topic.

it seems to me, that redirector is just miserable substitute :(

**Comment by Gitoffthelawn** [comment]
@alekksander Standard rules will work.  You just tell it what URLs to match and where to redirect them to.  You can create as many rules as meet your needs.

**Comment by jawz101** [comment]
I'm not a fan of the write-your-own-rules add-ons.  Something that pulls community rule sets, you train based on something which detects something and asks what you want to do.  Otherwise, the user has to constantly look at their address bar.

Also, I don't know if referrer and redirect are interchangeable terms but it seems like they are 2 separate things.

**Comment by Gitoffthelawn** [comment]
> Also, I don't know if referrer and redirect are interchangeable terms but it seems like they are 2 separate things.

You have it correct.  Not interchangeable terms.  Completely different things.

**Comment by Lootyhoof** [comment]
For what it's worth, this add-on does still work on [Pale Moon](http://www.palemoon.org/), and will continue to do so for the foreseeable future.

@diegocr If you want to have this hosted on an add-on repository the [Pale Moon Add-ons Site](https://addons.palemoon.org/) is available for any add-ons compatible with Pale Moon. If this is of interest feel free to ping me at the email address on my GitHub profile.

**Comment by Bluscream** [comment]
Can firefox download pale moon addons?

**Comment by Lootyhoof** [comment]
@Bluscream Yes, if you change the flag `xpinstall.signatures.required` to `false` in `about:config` first (and assuming any add-ons for Pale Moon also support Firefox). You wouldn't be able to auto update the add-on though, as auto update for the Pale Moon Add-ons Site only works in Pale Moon.

Also bearing in mind that this add-on will no longer work in Firefox versions past 57.

**Comment by Bluscream** [comment]
Why is mozilla enforcing so much bs even on experienced users :/

**Comment by LimboSlam** [comment]
> Why is mozilla enforcing so much bs even on experienced users :/

@Bluscream,  
 Have a look here: https://andreasgal.com/2017/05/25/chrome-won/, maybe this will give you some insight.

**Comment by RoestVrijStaal** [comment]
Hi, 

Because 2.7.1 isn't available here , I've extracted its XPI from my Pale Moon profile for the people who want version 2.7.1.

Simply unzip the zip and drag the XPI to your browser's window to install it.

[Clean Links 2.7.1-signed.1-signed.zip](https://github.com/diegocr/CleanLinks/files/1606589/Clean.Links.2.7.1-signed.1-signed.zip)
